SmartDeviceLink/NSMutableDictionary+Store.m

```diff
@@ -50,6 +53,13 @@ - (NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType {
     }
 }
 
+- (void)sdl_removeNullOccurrences {
+    NSArray<NSString *> *keys = [self allKeysForObject:[NSNull null]];
```

Comment:
This does not remove all `NSNull` values from a dictionary. All the responses from Core are `init`'d using `initWithDictionary:`. When you examine the `allKeysForObject` for the response it is a single key: `parameters`.  In order to find all the `NSNull` values, we must recursively search the dictionary for `NSNull` values. 

This is the `RegisterAppInterface` response with a `NSNull` value for `vrCapabilities`. The `vrCapabilities` is never removed from the dictionary in `sdl_removeNullOccurrences`  so when `vrCapabilities` is retrieved at a later point, the library crashes.  
```
{
    response =     {
        correlationID = 1;
        name = RegisterAppInterface;
        parameters =         {
            audioPassThruCapabilities =             (
                                {
                    audioType = PCM;
                    bitsPerSample = "8_BIT";
                    samplingRate = 44KHZ;
                }
            );
            buttonCapabilities =             (
                                {
                    longPressAvailable = 1;
                    name = "PRESET_0";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PRESET_1";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PRESET_2";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PRESET_3";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PRESET_4";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PRESET_5";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PRESET_6";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PRESET_7";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PRESET_8";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PRESET_9";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = OK;
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "PLAY_PAUSE";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = SEEKLEFT;
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = SEEKRIGHT;
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = TUNEUP;
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = TUNEDOWN;
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                },
                                {
                    longPressAvailable = 1;
                    name = "CUSTOM_BUTTON";
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                }
            );
            displayCapabilities =             {
                displayName = "SDL_HMI";
                displayType = "GEN2_8_DMA";
                graphicSupported = 1;
                imageFields =                 (
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = softButtonImage;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = choiceImage;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = choiceSecondaryImage;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = vrHelpItem;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = turnIcon;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = menuIcon;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = cmdIcon;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = graphic;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = secondaryGraphic;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = showConstantTBTIcon;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = showConstantTBTNextTurnIcon;
                    },
                                        {
                        imageResolution =                         {
                            resolutionHeight = 64;
                            resolutionWidth = 64;
                        };
                        imageTypeSupported =                         (
                            "GRAPHIC_BMP",
                            "GRAPHIC_JPEG",
                            "GRAPHIC_PNG"
                        );
                        name = showConstantTBTNextTurnIcon;
                    }
                );
                mediaClockFormats =                 (
                    CLOCK1,
                    CLOCK2,
                    CLOCK3,
                    CLOCKTEXT1,
                    CLOCKTEXT2,
                    CLOCKTEXT3,
                    CLOCKTEXT4
                );
                numCustomPresetsAvailable = 10;
                screenParams =                 {
                    resolution =                     {
                        resolutionHeight = 480;
                        resolutionWidth = 800;
                    };
                    touchEventAvailable =                     {
                        doublePressAvailable = 0;
                        multiTouchAvailable = 1;
                        pressAvailable = 1;
                    };
                };
                templatesAvailable =                 (
                    TEMPLATE
                );
                textFields =                 (
                                        {
                        characterSet = TYPE2SET;
                        name = mainField1;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = mainField2;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = mainField3;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = mainField4;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = statusBar;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = mediaClock;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = mediaTrack;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = alertText1;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = alertText2;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = alertText3;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = scrollableMessageBody;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = initialInteractionText;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = navigationText1;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = navigationText2;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = ETA;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = totalDistance;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = locationName;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = audioPassThruDisplayText1;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = audioPassThruDisplayText2;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = sliderHeader;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = sliderFooter;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = locationDescription;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = menuName;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = secondaryText;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = tertiaryText;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = 30;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = 31;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = menuTitle;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = addressLines;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = phoneNumber;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = 28;
                        rows = 1;
                        width = 500;
                    },
                                        {
                        characterSet = TYPE2SET;
                        name = 29;
                        rows = 1;
                        width = 500;
                    }
                );
            };
            hmiCapabilities =             {
                navigation = 1;
                phoneCall = 1;
                remoteControl = 1;
                videoStreaming = 1;
            };
            hmiDisplayLanguage = "EN-US";
            hmiZoneCapabilities =             (
                FRONT
            );
            iconResumed = 1;
            info = "";
            language = "EN-US";
            pcmStreamCapabilities =             {
                audioType = PCM;
                bitsPerSample = "16_BIT";
                samplingRate = 16KHZ;
            };
            prerecordedSpeech =             (
                "HELP_JINGLE",
                "INITIAL_JINGLE",
                "LISTEN_JINGLE",
                "POSITIVE_JINGLE",
                "NEGATIVE_JINGLE"
            );
            presetBankCapabilities =             {
                onScreenPresetsAvailable = 1;
            };
            resultCode = SUCCESS;
            sdlVersion = "{GIT_COMMIT}";
            softButtonCapabilities =             (
                                {
                    imageSupported = 1;
                    longPressAvailable = 1;
                    shortPressAvailable = 1;
                    upDownAvailable = 1;
                }
            );
            speechCapabilities =             (
                TEXT,
                "PRE_RECORDED",
                FILE
            );
            success = 1;
            supportedDiagModes =             (
                1,
                2,
                3,
                5,
                6,
                7,
                9,
                10,
                24,
                25,
                34,
                62
            );
            syncMsgVersion =             {
                majorVersion = 5;
                minorVersion = 0;
                patchVersion = 0;
            };
            systemSoftwareVersion = "ccpu_version";
            vehicleType =             {
                make = Ford;
                model = Fiesta;
                modelYear = 2013;
                trim = SE;
            };
            vrCapabilities = "<null>";
        };
    };
}
```